### DISCUSSION TITLE
PHP8.2 strtotime compliance

### DISCUSSION THREAD

**Discussion by aquavark** [squizlabs/PHP_CodeSniffer#3732] [Q&A] [answered]
Hi,

Is there a sniff that could detect, for PHP8.2, a correction that has been made in PHP strtotime that used to accept (wrongly as I've been told) the format date('Y-m-d',strtotime('2022-06-02 +-1 day')) which would give give 2022-06-01 in versions prior to 8.2, but in 8.2 gives 2022-06-03 (caught me out in some loops, the fix is to remove the + entirely as it's optional).

Many thanks

**Comment by jrfnl** [comment, marked as answer]
@aquavark As far as I know, no sniff exists for this (yet), but this would definitely be something of interest to add to the [PHPCompatibility](https://github.com/PHPCompatibility/PHPCompatibility) standard. You may want to open an issue about it there, preferably with a link to the specs of exactly what has changed.

**Comment by aquavark** [comment]
Thanks, will do!
